csharp/extractor/Semmle.Extraction.CIL/Context.cs

```diff
@@ -14,13 +14,19 @@ namespace Semmle.Extraction.CIL
     /// </summary>
     partial class Context : IDisposable
     {
-        public Extraction.Context cx;
         readonly FileStream stream;
-        public readonly MetadataReader mdReader;
-        public readonly PEReader peReader;
-        public readonly string assemblyPath;
-        public Entities.Assembly assembly;
-        public PDB.IPdb pdb;
+        Entities.Assembly? assemblyNull;
+
+        public Extraction.Context cx { get; }
+        public MetadataReader mdReader { get; }
+        public PEReader peReader { get; }
+        public string assemblyPath { get; }
+        public Entities.Assembly assembly
+        {
+            get { return assemblyNull!; }
+            set { assemblyNull = value; }
+        }
```

Comment:
I wonder if we could refactor this to avoid the `!`...

Reply:
I think this is the cleanest way in this case. We could refactor this a bit so that `Context` and `Assembly` have no circular reference to each other.